builders/build/substrate-api/polkadot-js-api.md

```diff
@@ -0,0 +1,298 @@
+
+```javascript
+// Subscribe to balance changes for our account
+const unsub = await api.query.system.account(ADDR, ({ nonce, data: balance }) => {
+  console.log(`free balance is ${balance.free} with ${balance.reserved} reserved and a nonce of ${nonce}`);
+});
+```
+
+## Keyrings {: #keyrings }
+
+The Keyring object is used for maintaining key pairs, and the signing of any data, whether it's a transfer, a message or a contract interaction.  
+
+### Creating a Keyring Instance {: #creating-a-keyring-instance }
+
+You can create an instance by just creating an instance of the Keyring class, and specifying the default type of wallet address used. For Moonbeam networks, the recommended default wallet type is `ethereum`.
+
+```javascript
+// Import the keyring as required
+import { Keyring } from '@polkadot/api';
+
+// Create a keyring instance
+const keyring = new Keyring({ type: 'ethereum' });
+```
+
+### Adding Accounts {: #adding-accounts }
+
+There are a number of ways to add an account to the keyring instance, including from the mnemonic phase, and from the shortform private key. The following sample code will provide some examples:
+
+```javascript
+// Import Ethereum Account from Mnemonic
+const keyringECDSA = new Keyring({ type: 'ethereum' });
+const mnemonic = 'mnemonic';
+
+// Define index of the derivation path and the derivation path
+const index = 0;
+const ethDerPath = "m/44'/60'/0'/0/" + index;
+const subsDerPath = '//hard/soft';
+console.log(`Mnemonic: ${mnemonic}`);
+console.log(`--------------------------\n`);
+
+// Extract Eth address from mnemonic
+const newPairEth = keyringECDSA.addFromUri(`${mnemonic}/${ethDerPath}`);
+console.log(`Ethereum Derivation Path: ${ethDerPath}`);
+console.log(`Derived Ethereum Address from Mnemonic: ${newPairEth.address}`);
+
+// Extract private key from mnemonic
+const privateKey = u8aToHex(
+  hdEthereum(mnemonicToLegacySeed(mnemonic, '', false, 64), ethDerPath).secretKey
+);
+console.log(`Derived Private Key from Mnemonic: ${privateKey}`);
+console.log(`--------------------------\n`);
+
+// Import Ethereum Account from Private Key
+// Define private key
+const privateKey = 'private_key';
+
+// Extract address from private key
+const otherPair = await keyringESDSA.addFromUri(privateKey);
+console.log(`Derived Address from Private Key: ${otherPair.address}`);
+```
+
+## Transactions {: #transactions }
+
+Transaction endpoints are exposed, as determined by the metadata, on the `api.tx` endpoint. These allow you to submit transactions for inclusion in blocks, be it transfers, deploying contracts, interacting with pallets, or anything else Moonbeam supports.
+
+### Sending Basic Transactions {: #sending-basic-transactions }
+
+Here is an example of sending a basic transaction from Alice to Bob:
+
+```javascript
+// Sign and send a transfer from Alice to Bob
+const txHash = await api.tx.balances
+  .transfer(BOB, 12345)
+  .signAndSend(alice);
```

Comment:
how can this be changed so users can just copy and paste this to have it work? like what are alice and bobs addresses? and also I think the format of alice and BOB should be the same, probably all lower case